Return-Path: <linux-block-owner@vger.kernel.org>
X-Original-To: lists+linux-block@lfdr.de
Delivered-To: lists+linux-block@lfdr.de
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.lfdr.de (Postfix) with ESMTP id DAA5614A921
	for <lists+linux-block@lfdr.de>; Mon, 27 Jan 2020 18:38:48 +0100 (CET)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1725908AbgA0Rir (ORCPT <rfc822;lists+linux-block@lfdr.de>);
        Mon, 27 Jan 2020 12:38:47 -0500
Received: from mx2.suse.de ([195.135.220.15]:37046 "EHLO mx2.suse.de"
        rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
        id S1725893AbgA0Rir (ORCPT <rfc822;linux-block@vger.kernel.org>);
        Mon, 27 Jan 2020 12:38:47 -0500
X-Virus-Scanned: by amavisd-new at test-mx.suse.de
Received: from relay2.suse.de (unknown [195.135.220.254])
        by mx2.suse.de (Postfix) with ESMTP id 0EB42AD21;
        Mon, 27 Jan 2020 17:38:46 +0000 (UTC)
Date:   Mon, 27 Jan 2020 18:38:45 +0100
From:   Daniel Wagner <dwagner@suse.de>
To:     Logan Gunthorpe <logang@deltatee.com>
Cc:     linux-block@vger.kernel.org,
        Johannes Thumshirn <Johannes.Thumshirn@wdc.com>,
        Omar Sandoval <osandov@fb.com>
Subject: Re: [PATCH blktests] nvme/018: Ignore message generated by nvme read
Message-ID: <20200127173845.5kbzrix524bmwwa6@beryllium.lan>
References: <20200127145353.52129-1-dwagner@suse.de>
 <a84f53ab-3f76-df73-e9f7-fb57c03eeb2d@deltatee.com>
MIME-Version: 1.0
Content-Type: text/plain; charset=us-ascii
Content-Disposition: inline
In-Reply-To: <a84f53ab-3f76-df73-e9f7-fb57c03eeb2d@deltatee.com>
Sender: linux-block-owner@vger.kernel.org
Precedence: bulk
List-ID: <linux-block.vger.kernel.org>
X-Mailing-List: linux-block@vger.kernel.org

On Mon, Jan 27, 2020 at 10:03:47AM -0700, Logan Gunthorpe wrote:
> > -	nvme read "/dev/${nvmedev}n1" -s "$sectors" -c 0 -z "$bs" 2>"$FULL" \
> > +	nvme read "/dev/${nvmedev}n1" -s "$sectors" -c 0 -z "$bs" &>"$FULL" \
> >  		&& echo "ERROR: Successfully read out of device lba range"
> 
> That's an odd message for an error... Error! I succeeded!

I think the idea is that the command should *not* succeed. So 'nvme
read' is expected to fail.

It is a bit confusing. What about something like

"ERROR: lba range read was unexpected successfull"?


